src/app/public/src/modules/nav/nav.component.spec.ts

```diff
@@ -121,6 +130,17 @@ describe('StacheNavComponent', () => {
     expect(mockRouter.navigate).toHaveBeenCalledWith(['/', 'test'], {});
   });
 
+  it('should navigate to an external url', () => {
+    spyOn(component, 'isExternal');
+    component.routes = [{ name: 'Test', path: 'https://google.com' }];
+    const route = component.routes[0];
+
+    fixture.detectChanges();
+    component.navigate(route);
+
+    expect(component['isExternal']).toHaveBeenCalled();
```

Comment:
So, I added a console log to the `if (this.isExternal())` block and it never gets called. I wonder if you could modify this test to spy on a mock window's `href` property, and check that instead?